rust/ip-packet/src/lib.rs

```diff
@@ -172,51 +157,64 @@ impl<'a> ConvertibleIpv4Packet<'a> {
         self.ip_header().destination_addr()
     }
 
-    fn consume_to_ipv6(
-        mut self,
-        src: Ipv6Addr,
-        dst: Ipv6Addr,
-    ) -> Option<ConvertibleIpv6Packet<'a>> {
-        let offset = nat46::translate_in_place(&mut self.buf, src, dst)
-            .inspect_err(|e| tracing::trace!("NAT64 failed: {e:#}"))
-            .ok()?;
-        let buf = self.buf.remove_from_head(offset);
+    fn consume_to_ipv6(mut self, src: Ipv6Addr, dst: Ipv6Addr) -> Option<ConvertibleIpv6Packet> {
+        dbg!(self.start);
+
+        // `translate_in_place` expects the packet to sit at a 20-byte offset.
+        // `self.start` tells us where the packet actually starts, thus we need to pass `self.start - 20` to the function.
+        let start_minus_padding = dbg!(self.start.checked_sub(20)?);
```

Comment:
here too

Reply:
if `checked_sub` fails we should panic here or not use `checked_sub`, this should never happen